cpp/src/arrow/flight/types.h

```diff
@@ -165,8 +166,15 @@ struct ARROW_FLIGHT_EXPORT BasicAuth {
   std::string username;
   std::string password;
 
+  /// \brief Deserialize this message from its wire-format representation.
+  static arrow::Result<BasicAuth> Deserialize(const arrow::util::string_view& serialized);
```

Comment:
Prefer passing by value (in registers) for trivial objects like `string_view`?
If passing by reference, the callee has to deference the pointer.

Reply:
Thanks for the catch - fixed.